lib/Controller/SieveController.php

```diff
@@ -109,7 +109,12 @@ public function updateActiveScript(int $id, string $script): JSONResponse {
 		$sieve = $this->getClient($id);
 
 		$scriptName = $sieve->getActive() ?? 'nextcloud';
-		$sieve->installScript($scriptName, $script, true);
+
+		try {
+			$sieve->installScript($scriptName, $script, true);
+		} catch (ManagesieveException $e) {
+			return new JSONResponse(statusCode: 422);
```

Comment:
inject the LoggerInterface and log the error as warning or error including the exception object

a day will come where we have to debug a script saving issue and then we'll be thankful to find information in the logs :)

Reply:
use the HTTP::STATUS_* constant for the status code